extension/json/json_functions/json_create.cpp

```diff
@@ -747,7 +747,7 @@ void JSONFunctions::RegisterJSONCreateCastFunctions(CastFunctionSet &casts) {
 			source_type = LogicalType::UNION({{"any", LogicalType::ANY}});
 			break;
 		case LogicalTypeId::ARRAY:
-			source_type = LogicalType::ARRAY(LogicalType::ANY);
+			source_type = LogicalType::ARRAYUNBOUNDED(LogicalType::ANY);
```

Comment:
`ARRAYUNBOUNDED` seems more confusing to me than the original version, as in this way it presents itself as a complete/usable type when it is an incomplete type only usable for binding. It's also inconsistent with how we handle other incomplete types (e.g. we just use `LogicalType::DECIMAL` instead of having a separate type for that).

Can we revert this change and potentially do address this in a different PR?

Reply:
Hmm that makes sense, the way I ran into this issue though is that it's easy to type ARRAY and accidentally leave out the array size, maybe we can instead make the array size a required argument, needs to be INVALID_INDEX to create an incomplete array type?